File: archive/merge-similar-items.js
const mergeSimilarItems = function(items1, items2) {
  let mem = {}

  items2.forEach(el => {
    if (mem[el[0]]===undefined) {
      mem[el[0]] = el[1]
    }
  })

  items1.forEach((el) => {
    if (mem[el[0]]===undefined) {
      mem[el[0]] = el[1]
    } else {
      mem[el[0]]+=el[1]
    }
  })

  let ans = []

  Object.keys(mem).forEach(el => {
    ans.push([Number(el), mem[el]])
  })
  ans.sort((a,b) => a[0]-b[0])
  return ans
};

console.log(mergeSimilarItems([[1,1],[4,5],[3,8]], [[3,1],[1,5]]))

File: merge-similar-items.go
func mergeSimilarItems(items1 [][]int, items2 [][]int) [][]int {
    mem:=map[int]int{}

    items := append(items1, items2...)

    for i:=0; i < len(items); i++ {
        if _, isHas := mem[items[i][0]]; isHas {
            mem[items[i][0]]+=items[i][1]
        } else {
            mem[items[i][0]] = items[i][1]
        }
    }

    ans:=[][]int{}

    for k,v := range mem {
        ans = append(ans, []int{k, v})
    }

    sort.SliceStable(ans, func (i int, j int) bool {
        return ans[i][0] < ans[j][0]
    })

    return ans
}